packages/atlas/src/hooks/useWaitForFee.ts

```diff
@@ -0,0 +1,47 @@
+import BN from 'bn.js'
+import { useCallback, useEffect, useState } from 'react'
+
+import { useSubscribeAccountBalance } from '@/providers/joystream/joystream.hooks'
+import { useSnackbar } from '@/providers/snackbars'
+
+export const useWaitForFee = (feeLoading: boolean, fee?: BN, callback?: () => void) => {
+  const [loadingState, setLoadingState] = useState(false)
+  const { totalBalanceLoaded, totalBalance } = useSubscribeAccountBalance()
+  const { displaySnackbar } = useSnackbar()
+
+  const callbackHandler = useCallback(() => {
+    if (!fee) {
+      return
+    }
+    if (totalBalance?.lt(fee)) {
+      displaySnackbar({
+        title: 'Not enough funds',
+        description:
+          "You don't have enough funds to cover blockchain transaction fee for this operation. Please, top up your account balance and try again.",
+        iconType: 'error',
+      })
+      return
+    }
+    callback?.()
+  }, [callback, displaySnackbar, fee, totalBalance])
+
+  useEffect(() => {
+    if (loadingState && !feeLoading && totalBalanceLoaded) {
+      callbackHandler?.()
+      setLoadingState(false)
+    }
+  }, [callback, loadingState, feeLoading, totalBalanceLoaded, totalBalance, displaySnackbar, callbackHandler])
+
+  const actionHandler = useCallback(() => {
```

Comment:
this is really unintuitive name. If you see something like that in the component like `Button onClick={actionHandler}` this means nothing to you basically. Can we rename the function name to something more associated with `fee`?